src/data_readers/data_reader_image.cpp

```diff
@@ -119,7 +119,15 @@ void image_data_reader::set_input_params(const int width, const int height, cons
 
 bool image_data_reader::fetch_label(CPUMat& Y, int data_id, int mb_idx) {
   const label_t label = m_image_list[data_id].second;
-  Y.Set(label, mb_idx, 1);
+  if (label >= 0 && label < m_num_labels) {
```

Comment:
I would use static_cast<label_t>(0) and static_cast<label_T>(m_num_labels) just in case.

Reply:
Done